internals/src/macros.rs

```diff
@@ -139,8 +139,11 @@ macro_rules! debug_from_display {
 /// Asserts a boolean expression at compile time.
 #[macro_export]
 macro_rules! const_assert {
-    ($x:expr) => {
-        const _: [(); 0 - !$x as usize] = [];
+    ($x:expr $(; $message:expr)?) => {
+        if !$x {
+            // We can't use formatting in const, only concating literals.
+            panic!(concat!("assertion ", stringify!($x), " failed" $(, ": ", $message)?))
+        }
```

Comment:
This is broken because previously it expanded to `const _`, now it doesn't.

Reply:
Oh, nice. Now we can get rid of the module again.